instrumentation/net/http/otelhttp/wrap.go

```diff
@@ -86,10 +86,9 @@ func (w *respWriterWrapper) Write(p []byte) (int, error) {
 }
 
 func (w *respWriterWrapper) WriteHeader(statusCode int) {
-	if w.wroteHeader {
-		return
+	if !w.wroteHeader {
+		w.wroteHeader = true
+		w.statusCode = statusCode
```

Comment:
If feel that this change is not correct.

You always call `w.ResponseWriter.WriteHeader(statusCode)` but on the other side you assume that it is not possible to change the status code after it was initially set (becasuse `w.statusCode = statusCode` is inside an `if` statement).

If there would be a `ResponseWriter` that allows setting the status code multiple times then this PR does not fix it.

Personally, I would rather consider changing this code to:

```go
func (w *respWriterWrapper) WriteHeader(statusCode int) {
	w.statusCode = statusCode
	w.ResponseWriter.WriteHeader(statusCode)
	w.wroteHeader = true
}
```

but then it would lead to strange behavior when the default `ResponseWriter` is used (but there would be logs from http package)...

Reply:
Per https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3580#issuecomment-1468123160

I think that it would be good enough to just leave a comment describing the behavior in more detail e.g. to avoid a regression bug.